libgadget/blackhole.c

```diff
@@ -514,17 +830,42 @@ blackhole_accretion_postprocess(int i, TreeWalk * tw)
     double dtime = get_dloga_for_bin(P[i].TimeBin) / All.cf.hubble;
 
     BHP(i).Mass += BHP(i).Mdot * dtime;
+
+    /*************************************************************************/
+    
+    if(blackhole_params.BH_DRAG > 0){
+        double fac = 0;
+        if (blackhole_params.BH_DRAG == 1) fac = BHP(i).Mdot/P[i].Mass; 
+        if (blackhole_params.BH_DRAG == 2) fac = blackhole_params.BlackHoleEddingtonFactor * meddington/BHP(i).Mass;
+        for(k = 0; k < 3; k++) {
+            fac *= All.cf.a; /* dv = acc * kick_fac = acc * a^{-1}dt, therefore acc = a*dv/dt  */
+            BHP(i).DragAccel[k] = -(P[i].Vel[k] - BH_GET_PRIV(tw)->BH_SurroundingGasVel[PI][k])*fac;
+            P[i].GravAccel[k]  += BHP(i).DragAccel[k]; 
```

Comment:
Instead of adding this to GravAccel here, why not add DragAccel and DFAccel inside the do_the_short_range_kick() in timestep.c like we do for HydroAccel. So inside do_the_short_range_kick you would have:
if(Type == 5) {
Vel += BHP(i).DragAccel[k];
}
(with the appropriate factors)